content/exsaml.xqm

```diff
@@ -123,30 +130,34 @@ declare %private function exsaml:build-saml-authnreq() {
 };
 
 declare %private function exsaml:store-authnreqid-as-exsol-user($id as xs:string, $instant as xs:string) {
+      let $saml-coll-reqid := $exsaml:saml-coll-reqid-base || "/" || $exsaml:saml-coll-reqid-name
       let $create-collection := 
         if (        
-            not(xmldb:collection-available($exsaml:saml-coll-reqid))
+            not(xmldb:collection-available($saml-coll-reqid))
         )
         then (
-            let $log := exsaml:log("info", "collection " || $exsaml:saml-coll-reqid || " does not exist, attempting to create it")
+            let $log := exsaml:log("info", "collection " || $saml-coll-reqid || " does not exist, attempting to create it")
             return
-                xmldb:create-collection("/db/apps/existdb-saml", "saml-request-ids")
+                xmldb:create-collection($exsaml:saml-coll-reqid-base, $exsaml:saml-coll-reqid-name)
         )
         else ()
     return
-        xmldb:store($exsaml:saml-coll-reqid, $id, <reqid>{$instant}</reqid>)
+        xmldb:store($saml-coll-reqid, $id, <reqid>{$instant}</reqid>)
   
 };
 
-(: store issued request ids in a collection,  :)
+(: store issued request ids in a collection :)
+(: this is pointless if we are configured to accept unsolicited SAML assertions :)
 declare %private function exsaml:store-authnreqid($id as xs:string, $instant as xs:string) {
-    let $log := exsaml:log("info", "storing SAML request id: " || $id || ", date: " || $instant)
-    return
-        system:as-user(
+    if (not($exsaml:idp-unsolicited)) then (
+        let $log := exsaml:log("info", "storing SAML request id: " || $id || ", date: " || $instant)
```

Comment:
If we combine the calls to exams:log and system:as-user we can get rid of the return statement.

```
(
    exsaml:log("info", "storing SAML request id: " || $id || ", date: " || $instant),
    system:as-user(
        $exsaml:exsaml-user,
        $exsaml:exsaml-pass,
        exsaml:store-authnreqid-as-exsol-user($id, $instant)
    )
)
```

Reply:
The above is only possible if the log method returns an empty sequence instead of `true()`
https://github.com/eXist-db/existdb-saml/blob/9c43d19ec2eea1c26e749952f0c1681a7c889b18/content/exsaml.xqm#L636

Reply:
the log method returns true on purpose, so no.